datafusion/physical-expr/src/equivalence/properties.rs

```diff
@@ -2617,379 +2633,356 @@ mod tests {
         ))
     }
 
-    #[tokio::test]
-    async fn test_union_equivalence_properties_multi_children() -> Result<()> {
-        let schema = create_test_schema()?;
+    #[test]
+    fn test_union_equivalence_properties_multi_children_1() {
+        let schema = create_test_schema().unwrap();
         let schema2 = append_fields(&schema, "1");
         let schema3 = append_fields(&schema, "2");
-        let test_cases = vec![
-            // --------- TEST CASE 1 ----------
-            (
+        UnionEquivalenceTest::new(&schema)
```

Comment:
I wanted more flexibility with the tests, and I am glad I did this as I hadn't seen the existing tests for constants